testsuite/src/Test/Tasty/Clash.hs

```diff
@@ -269,15 +273,17 @@ instance IsTest ClashTest where
         (testExitCode testExit) "clash" (args oDir) NoGlob PrintNeither False
         (specificExitCode testExit) (ExpectStdErr expectedErr) Nothing
 
+    modFlags = if ctLoadFromInterface
+                 then [ctModName]
+                 else ["-i" <> ctSourceDirectory
+                      , ctSourceDirectory </> ctModName <.> "hs"]
```

Comment:
I don't think you even need the switch. Could you test if the following works:

1. Always use the `-i` stuff
2. Always call a module by its name, not its file

Loading from interface files should take precedence over loading from local files. If that works we could simply remove the flag from the options record.

Reply:
Actually scratch that